File: CountAndSay/CountAndSay.py
class Solution(object):
    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """
        # result = []
        # tmp = str(n) + '0'
        # count = 1
        # for i in range(len(tmp)-1):
        #     if tmp[i] == tmp[i+1]:
        #         count += 1
        #     else:
        #         result.append(str(count))
        #         result.append(tmp[i])
        #         count = 1
        # return ''.join(result)
        res = []
        res.append('1')
        for i in range(n-1):
            res.append(self.create(res[i]))
        return res[n-1]
        
        
    def create(self, str1):
        tmp = str1 + '0'
        result = []
        count = 1
        for i in range(len(tmp)-1):
            if tmp[i] == tmp[i+1]:
                count += 1
            else:
                result.append(str(count))
                result.append(tmp[i])
                count = 1
        return ''.join(result)

File: CountAndSay/CountAndSay.cpp
#include<string>
#include<vector>
#include<algorithm>
using namespace std;

class Solution {
public:
	string countAndSay(int n) {
		string res = "1";
		if (n <= 0)
			return "";
		for (int i = 0; i < n - 1; i++) {
			string tmp = "";
			int len = res.size();
			for (int j = 0; j < len; j++) {
				int count = 1;
				while ((j + 1 < len) && res[j] == res[j + 1]){
					count++;
					j++;
				}
				tmp += to_string(count) + res[j];
			}
			res = tmp;
		}
		return res;
	}
};